api/metrics/taxitrips.py

```diff
@@ -0,0 +1,35 @@
+from api.utils.database import rows_to_dicts
+
+#multiple metrics for the taxi trip dataset
+class TaxiTripMetrics:
+    def __init__(self, connection):
+        self.connection = connection
+        
+        
+#gets the average speed of a taxi 
+    def avgtaxispeed(self):
+        query = """
+        SELECT
+            trip_miles,
+            trip_minutes/60 as trip_hours,
+            trip_miles/trip_hours as trip_speed
+        FROM taxitrips
+        GROUP BY taxi_id
+        """
+        cur = self.connection.cursor()
+        cur.execute(query)
+        rows = rows_to_dicts(cur, cur.fetchall())
+        return rows
+        
+#takes in pickup_community_area and returns most common dropoff_community_area
+    def maxdropoff(self, pickup_community_area):
+        query = """
+        SELECT max(dropoff_community_area)
```

Comment:
The max() function will return the largest value, not the most common value.

Happy to pair program with you on how to get the most common value, or you can Google something like this:

"SQL get most common value in group by"

This would be a good metric to have for a data question, but it's not the kind of metric we can show on the timeline or community scatter view. It would be great to have a metric that is total number of trips by pickup area and another by dropoff area. If you return two columns: one called area_number and one called value, that metric will plug right into our community scatter plot. You could also do a total trips per day metric, which should have one column called date and one called value to plug into the timeline view.

Let me know if anything I said above is unclear, I can show you how the metrics plug into the frontend.